### PR TITLE
Util methods ts 

### PR DESCRIPTION
**Pull request by 0xgabrielp**

### CODE REVIEW

File: src/types/asset-allocation.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import BN from "bn.js";
+
+export interface AssetAllocation  {
+    assetGuid: string;
+    values: {
+        n: number;
+        value: BN;
+    }[];
+    notarysig: string | Buffer;
+}[]
```
Comment (osiastedian) [inline src/types/asset-allocation.ts]: Theres an  unknown `[]` here.

**Comment by osiastedian** [comment]
Screen Shot 2022-07-04 at 10 26 46 AM
There are formatting issues

**Comment by osiastedian** [comment]
Screen Shot 2022-07-04 at 10 28 23 AM
There's also lint issues.

File: src/types/notary-details.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface NotaryDetails {
+    endpoint: string | Buffer;
+    instanttransfers: boolean,
+    hdrequired: boolean
+}
+
+export interface NotaryAsset {
```
Comment (osiastedian) [inline src/types/notary-details.ts]: You can extend `Record<string, any>`

File: src/utils/functions/fetchBackendUTXOS.ts [hunk 3]
```diff
@@ -1,5 +1,16 @@
 import axios from 'axios';
-import { UtxoObject } from '../../types/utxo-object';
+
+/**
+ * Refer to https://github.com/trezor/blockbook/blob/master/docs/api.md#get-utxo
+ */
+interface UTXO {
```
Comment (osiastedian) [inline src/utils/functions/fetchBackendUTXOS.ts]: I think you can update the existing interface found on 

```
src/types/utxo-object.ts
```

File: src/utils/functions/getAllocationsFromTx.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import syscointx from 'syscointx-js';
+import bjs from 'bitcoinjs-lib'
+import { AssetAllocation } from '../../types/asset-allocation';
+
+/**
+ * Purpose: Return allocation information for an asset transaction. Pass through to syscointx-js
+ * @param tx Required. bitcoinjs transaction
+ * @returns allocation information
+ */
+export function getAllocationsFromTx (tx: bjs.Transaction): AssetAllocation {
```
Comment (osiastedian) [inline src/utils/functions/getAllocationsFromTx.ts]:
```typescript
export function getAllocationsFromTx (tx: bjs.Transaction): AssetAllocation [] {
```

File: src/utils/functions/getNotarizationSignatures.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { NotaryAsset } from "../../types/notary-details"
+import fetchNotarizationFromEndPoint from "./fetchNotarizationFromEndPoint"
+
+/**
+ * Get notarization signatures from a notary endpoint defined in the asset object, see spec for more info: https://github.com/syscoin/sips/blob/master/sip-0002.mediawiki
+ * @param notaryAssets Required. Asset objects that require notarization, fetch signatures via fetchNotarizationFromEndPoint()
+ * @param txHex Required. Signed transaction hex created from syscointx.createTransaction()/syscointx.createAssetTransaction()
+ * @returns boolean representing if notarization was done by acquiring a witness signature from notary.
+ */
+export async function getNotarizationSignatures (notaryAssets: Map<string, NotaryAsset>, txHex: string): Promise<boolean> {
```
Comment (osiastedian) [inline src/utils/functions/getNotarizationSignatures.ts]: Lets temporarily add ESLint disables as we don't want to alter the logic for this yet.

```typescript
/* eslint-disable no-await-in-loop */
/* eslint-disable no-continue */
/* eslint-disable no-restricted-syntax */
```

File: src/utils/functions/notarizePSBT.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ * Purpose: Notarize Result object from syscointx.createTransaction()/syscointx.createAssetTransaction() if required by the assets in the inputs of the transaction
+ * @param psbt Required. The resulting PSBT object passed in which is assigned from syscointx.createTransaction()/syscointx.createAssetTransaction()
+ * @param notaryAssets Required. Asset objects require notarization, fetch signatures via fetchNotarizationFromEndPoint()
+ * @param rawTx Required. Transaction hex
+ * @returns 
+ */
+export async function notarizePSBT (psbt: Psbt, notaryAssets: Map<string, NotaryAsset>, rawTx: string): Promise<boolean> {
+    const notarizationDone = await getNotarizationSignatures(notaryAssets, rawTx)
+    if (notarizationDone) {
+      return syscointx.addNotarizationSignatures(psbt.version, notaryAssets, psbt.txOutputs)
```
Comment (osiastedian) [inline src/utils/functions/notarizePSBT.ts]: Kindly check the return type of `addNotarizationSignatures`.
Should be available on `node_modules/syscointx-js`

**Review by osiastedian — CHANGES_REQUESTED** [review]

File: src/utils/functions/notarizePSBT.ts (hunk 6 above)
Comment (0xgabrielp) [reply src/utils/functions/notarizePSBT.ts]: Added the type `{ output: any; index: number }`, 
found it on https://github.com/syscoin/syscointx-js/blob/master/index.js#L187

**Review by osiastedian — APPROVED** [review]

**Pull request merged**